platform/security/interceptor/security-interceptor-anonymous/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptor.java

```diff
@@ -445,23 +451,25 @@ private void createSecurityToken(SoapVersion version, SOAPFactory soapFactory,
         }
     }
 
-    private Subject getSubject() {
-        if (Security.tokenAboutToExpire(cachedAnonymousSubject)) {
+    private Subject getSubject(String ipAddress) {
+        Subject subject = cachedAnonymousSubjectMap.get(ipAddress);
```

Comment:
I believe you'd need this get to also by synchronized, since the "put" is synchronized.